synapse/models/geospace.py

```diff
@@ -1,6 +1,66 @@
+import synapse.lib.gis as s_gis
+import synapse.lib.types as s_types
+import synapse.lib.syntax as s_syntax
+
 from synapse.lib.module import CoreModule
 
-latlongre = '^[-+]?([1-8]?\d(\.\d+)?|90(\.0+)?),\s*[-+]?(180(\.0+)?|((1[0-7]\d)|([1-9]?\d))(\.\d+)?)$'
+class LatLongType(s_types.DataType):
+
+    def norm(self, valu, oldval=None):
+
+        valu = valu.strip().lower()
+        if valu == '??':
+            return valu, {}
+
+        try:
+
+            lat, lon = valu.split(',', 1)
+
+            latv = float(lat.strip())
+            lonv = float(lon.strip())
+
+        except Exception as e:
+            self._raiseBadValu(valu, mesg='Invalid float format')
+
+        #TODO eventually support minutes / sec and N/S E/W syntax
+
+        if latv > 90 or latv < -90:
+            self._raiseBadValu(valu, mesg='Latitude may only be -90.0 to 90.0')
+
+        if lonv > 180 or lonv < -180:
+            self._raiseBadValu(valu, mesg='Longitude may only be -180.0 to 180.0')
+
+        norm = '%s,%s' % (latv, lonv)
+        return norm, {'lat': latv, 'lon': lonv}
+
+units = {
+    'mm': 1,
+    'cm': 10,
+
+    'm': 1000,
+    'meters': 1000,
+
+    'km': 1000000,
+}
+
+class DistType(s_types.DataType):
+
+    def norm(self, valu, oldval=None):
+
+        if type(valu) == str:
+            return self._norm_str(valu)
+        return valu, {}
+
+    def _norm_str(self, text):
+
+        valu, off = s_syntax.parse_float(text, 0)
+        unit, off = s_syntax.nom(text, off, s_syntax.alphaset)
+
+        mult = units.get(unit.lower())
+        if mult is None:
+            raise SyntaxError('invalid units: %s' % (unit,))
```

Comment:
Should this `SyntaxError` be `BadTypeValu`?

Reply:
good call.